eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -154,7 +158,7 @@ trait Eclair {
 
   def verifyMessage(message: ByteVector, recoverableSignature: ByteVector): VerifiedMessage
 
-  def sendOnionMessage(route: Seq[PublicKey], userCustomContent: ByteVector, pathId: Option[ByteVector])(implicit timeout: Timeout): Future[SendOnionMessageResponse]
+  def sendOnionMessage(intermediateNodes: Seq[OnionMessages.IntermediateNode], destination: Either[OnionMessages.Recipient, Sphinx.RouteBlinding.BlindedRoute], replyPath: Option[Seq[PublicKey]], userCustomContent: ByteVector)(implicit timeout: Timeout): Future[SendOnionMessageResponse]
```

Comment:
This function should avoid using business logic types unless really necessary, it should stay as general as possible (and avoid needing to change when business logic types change): callers don't need to know about the `OnionMessages.*` types in most cases, a simple `PublicKey` is enough.

I think we can use simplify it to the following definition and create the business logic types inside of the function body:
```suggestion
  def sendOnionMessage(intermediateNodes: Seq[PublicKey], destination: Either[PublicKey, Sphinx.RouteBlinding.BlindedRoute], replyPath: Option[Seq[PublicKey]], userCustomContent: ByteVector)(implicit timeout: Timeout): Future[SendOnionMessageResponse]
```